### PR TITLE
Add config for additional conf.d and checks.d volume mounts

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

Allow configuring `conf.d` and `checks.d` volume mounts by providing config maps defined by user.

### Motivation

Users would like to provide their custom config files (`conf.d`) and integration checks (`checks.d`).

### Additional Notes

This is still a WIP to solicit early feedback.

### CODE REVIEW

File: pkg/controller/datadogagentdeployment/agent_test.go [hunk 1]
```diff
@@ -342,7 +466,7 @@ func Test_newExtendedDaemonSetFromInstance(t *testing.T) {
 				return
 			}
 			if !apiequality.Semantic.DeepEqual(got, tt.want) {
-				t.Errorf("newExtendedDaemonSetFromInstance() = %#v, want %#v", got, tt.want)
+				t.Errorf("newExtendedDaemonSetFromInstance() = %#v\n\nwant %#v\ndiff: %s", got, tt.want, cmp.Diff(got, tt.want))
```
Comment (ahmed-mez) [inline pkg/controller/datadogagentdeployment/agent_test.go]: 👍

File: pkg/controller/datadogagentdeployment/agent_test.go [hunk 2]
```diff
@@ -35,6 +37,10 @@ func Test_newExtendedDaemonSetFromInstance(t *testing.T) {
 			Name:      "confd",
 			MountPath: "/conf.d",
 		},
+		{
+			Name:      "checksd",
+			MountPath: "/checks.d",
```
Comment (ahmed-mez) [inline pkg/controller/datadogagentdeployment/agent_test.go]: we might want to add `ReadOnly:  true,` here - same for `confd`
( https://github.com/helm/charts/blob/master/stable/datadog/templates/container-agents.yaml#L170 )

File: pkg/controller/datadogagentdeployment/utils.go [hunk 3]
```diff
@@ -635,6 +662,10 @@ func getVolumeMountsForAgent(spec *datadoghqv1alpha1.DatadogAgentDeploymentSpec)
 			Name:      datadoghqv1alpha1.ConfdVolumeName,
 			MountPath: datadoghqv1alpha1.ConfdVolumePath,
 		},
+		{
+			Name:      datadoghqv1alpha1.ChecksdVolumeName,
+			MountPath: datadoghqv1alpha1.ChecksdVolumePath,
```
Comment (ahmed-mez) [inline pkg/controller/datadogagentdeployment/utils.go]: we might want to add `ReadOnly:  true,` here - same for `confd`
( [helm chart](https://github.com/helm/charts/blob/master/stable/datadog/templates/container-agents.yaml#L170) )

File: pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go [hunk 4]
```diff
@@ -114,6 +114,14 @@ type DatadogAgentDeploymentSpecAgentSpec struct {
 	// SystemProbe configuration
 	// +optional
 	SystemProbe SystemProbeSpec `json:"systemProbe,omitempty"`
+
+	// Confd configuration
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go]: could you add more documentation / context here based on https://docs.datadoghq.com/agent/guide/agent-configuration-files 🙏

File: pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go [hunk 5]
```diff
@@ -114,6 +114,14 @@ type DatadogAgentDeploymentSpecAgentSpec struct {
...
+	// +optional
+	Confd *DirConfig `json:"confd,omitempty"`
+
+	// Checksd configuration
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go]: could you add more documentation here as well ? https://docs.datadoghq.com/developers/write_agent_check

File: pkg/controller/datadogagentdeployment/utils.go (hunk 3 above)
Comment (xornivore) [reply pkg/controller/datadogagentdeployment/utils.go]: Good point, thank you! Will update to be `ReadOnly`.

File: vendor/modules.txt [hunk 6]
```diff
@@ -13,12 +13,15 @@ github.com/PuerkitoBio/purell
 github.com/PuerkitoBio/urlesc
 # github.com/beorn7/perks v1.0.0
 github.com/beorn7/perks/quantile
-# github.com/datadog/extendeddaemonset v0.1.6 => ../ddaemonset/src/github.com/datadog/extendeddaemonset
+# github.com/datadog/extendeddaemonset v0.1.0
```
Comment (xornivore) [inline vendor/modules.txt]: This probably needs to be reverted... not sure why this change.

Comment (xornivore) [reply vendor/modules.txt]: Discussed with @clamoriniere this is actually correct, should be `v0.1.0`.

File: test/e2e/datadogagentdeployment_test.go [hunk 7]
```diff
@@ -234,6 +235,80 @@ func DeploymentDaemonset(t *testing.T) {
...
+		if pod.Status.Phase != corev1.PodRunning {
+			t.Logf("pod %s is not Running (yet?)", name)
+			return false, nil
+		}
+
+		for _, vol := range pod.Spec.Volumes {
+			if vol.ConfigMap != nil && needConfigMaps[vol.ConfigMap.Name] {
+				count++
+			}
+			if count == len(needConfigMaps) {
```
[the remark this answers is not in the document]
Comment (xornivore) [reply test/e2e/datadogagentdeployment_test.go]: Ah yes, since we need all pods.

File: vendor/modules.txt (hunk 6 above)
Comment (clamoriniere) [reply vendor/modules.txt]: Yes sorry I have committed some local change...

File: test/e2e/datadogagentdeployment_test.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (xornivore) [reply test/e2e/datadogagentdeployment_test.go]: Actually just discussed - this allows breaking early out of the loop for volumes.
